### ISSUE TITLE
OpenTelemetry source level automatic instrumentation addition (donation)

### ISSUE THREAD

**Issue by pdelewski** [open-telemetry/community#1105]

**Comment by tigrannajaryan** [comment]
@pdelewski thank you for the proposal. I will add this to the next TC meeting to discuss.

Just to confirm: this is an independent effort, unrelated to https://github.com/open-telemetry/community/issues/1055

**Comment by pdelewski** [comment]
> @pdelewski thank you for the proposal. I will add this to the next TC meeting to discuss.
> 
> Just to confirm: this is an independent effort, unrelated to #1055 ?

Yes, it's independent

**Comment by yurishkuro** [comment]
Please provide answers to all questions asked in the form, specifically:
  * Summary of existing usage of donated project
  * Maintenance plan

Also, please explain why you want to donate this code, what goals you want to achieve ("we don't want to support it" is a valid goal, but...), and why those goals are easier achieved by the project being in OTEL rather than in a standalone repo.

**Comment by jmacd** [comment]
@pdelewski See my comments in the _parallel_ donation proposal here:
https://github.com/open-telemetry/community/issues/1055#issuecomment-1199872692

@pdelewski are you willing to take ownership of the code that you propose donating and eventually become a maintainer of the (proposed) new opentelemetry-go-instrumentation repository?

**Comment by pdelewski** [comment]
@jmacd Yes, I agree  to take ownership of the proposed code and I will be happy to become maintainer of new repo.

Thank you for  feedback and review.

**Comment by jmacd** [comment]
@pdelewski I created https://github.com/open-telemetry/opentelemetry-go-instrumentation. One of the first tasks for you and @edenFed will be to become members of the OpenTelemetry org and begin sending and reviewing PRs in this repository. I copied the current Go maintainers in as maintainers of this project temporarily; please consider me your sponsor, I will be glad to help review the first steps in this repository as we work to establish you as new a new maintainer.
